tests/test_viewing.py

```diff
@@ -304,7 +300,331 @@ def test_view_attrs(module):
+            for comp in branch.comps:
+                nodes1.append(comp.nodes)
+    assert len(nodes1) == len(net.nodes), "Some compartments were skipped in iteration."
+
+    nodes2 = []
+    for cell in net:
+        for branch in cell:
+            for comp in branch:
+                nodes2.append(comp.nodes)
+    assert len(nodes2) == len(net.nodes), "Some compartments were skipped in iteration."
+    assert np.all(
+        [np.all(n1 == n2) for n1, n2 in zip(nodes1, nodes2)]
+    ), "__iter__ is not consistent with [comp.nodes for cell in net.cells for branches in cell.branches for comp in branches.comps]"
+
+    assert np.all(
+        [len(comp.nodes) for comp in net[0, 0].comps] == [1, 1]
+    ), "Iterator yielded unexpected number of compartments"
+
+    # 0th comp in every branch (3), 1st comp in every branch (3), 2nd comp in (every) branch (only 1 branch with > 2 comps)
+    assert np.all(
+        [len(comp.nodes) for comp in net[0].comps] == [3, 3, 1]
+    ), "Iterator yielded unexpected number of compartments"
+
+    # 0th comp in every branch for every cell (6), 1st comp in every branch for every cell , 2nd comp in (every) branch for every cell
+    assert np.all(
+        [len(comp.nodes) for comp in net.comps] == [6, 6, 2]
+    ), "Iterator yielded unexpected number of compartments"
+
+    for comp in branch1:
+        comp.set("v", -72)
+    assert np.all(branch1.nodes["v"] == -72), "Setting parameters with __iter__ failed."
+
+    # needs to be redefined because cell was overwritten with View object
+    cell = jx.Cell([branch1, branch1, branch2], parents=[-1, 0, 0])
+    for branch in cell:
+        for comp in branch:
+            comp.set("v", -73)
+    assert np.all(cell.nodes["v"] == -73), "Setting parameters with __iter__ failed."
+
+
+def test_synapse_and_channel_filtering():
+    """Test that synapses and channels are filtered correctly by View."""
+    comp = jx.Compartment()
+    branch = jx.Branch([comp] * 3)
+    cell = jx.Cell([branch] * 3, parents=[-1, 0, 0])
+    net = jx.Network([cell] * 3)
+    net.insert(HH())
+    connect(net[0, 0, :], net[1, 0, :], TestSynapse())
+
+    assert np.all(net.cell(0).HH.nodes == net.HH.cell(0).nodes)
+    view1 = net.cell([0, 1]).TestSynapse
+    nodes1 = view1.nodes
+    edges1 = view1.edges
+    view2 = net.TestSynapse.cell([0, 1])
+    nodes2 = view2.nodes
+    edges2 = view2.edges
+    nodes_control_param1 = nodes1.pop("controlled_by_param")
+    nodes_control_param2 = nodes2.pop("controlled_by_param")
+    edges_control_param1 = edges1.pop("controlled_by_param")
+    edges_control_param2 = edges2.pop("controlled_by_param")
+
+    # convert to dict so order of cols and index dont matter for __eq__
+    assert nodes1.to_dict() == nodes2.to_dict()
+    assert np.all(nodes_control_param1 == 0)
+    assert np.all(nodes_control_param2 == nodes2["global_cell_index"])
+
+    assert np.all(edges1 == edges2)
+
+
+def test_view_equals_module():
```

Comment:
add test to check if `comp` and `branch.comp(0)` behave the same for recordings, setting params and stimulating